### PR TITLE
[DATAPOR-35] Ckan user managment integration

### PR DESCRIPTION
**Pull request by acherici**

### CODE REVIEW

File: catalog_manager/app/it/gov/daf/catalogmanager/utilities/SecurePasswordHashing.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    if (hash.right.get.length == 0 || salt.right.get.length == 0) return false
+
+    val calculatedHash = pbkdf2(password, salt.right.get, nrOfIterations)
+
+    slowEquals(calculatedHash, hash.right.get)
+  }
+
+  private def pbkdf2(password: String, salt: Array[Byte], nrOfIterations: Int): Array[Byte] = {
+    val keySpec = new PBEKeySpec(password.toCharArray(), salt, nrOfIterations, SizeOfPasswordHashInBytes * 8)
+    val keyFactory = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA1")
```
Comment (gvarisco) [inline catalog_manager/app/it/gov/daf/catalogmanager/utilities/SecurePasswordHashing.scala]: We should be probably having a look at PBKDF2 with HMAC-SHA256 or HMAC-SHA512. AFAIK they are only available on JDK8 or with a custom JCE provider such as IAIK or BSAFE.

Comment (gvarisco) [inline catalog_manager/app/it/gov/daf/catalogmanager/utilities/SecurePasswordHashing.scala]: there's a good explanation here: https://netnix.org/2015/04/19/aes-encryption-with-hmac-integrity-in-java/

Comment (gvarisco) [inline catalog_manager/app/it/gov/daf/catalogmanager/utilities/SecurePasswordHashing.scala]: some performance tests/comparisons: http://www.everit.org/password-encryptor-pbkdf2/

Comment (giux78) [inline catalog_manager/app/it/gov/daf/catalogmanager/utilities/SecurePasswordHashing.scala]: thanks!!!

**Pull request merged**